### PR TITLE
Update disaggregation_LUH2.R

### PR DESCRIPTION
**Pull request by emolinab**
## :bird: Description of this PR :bird:

- In this update, we fix the definition of what "flooded" area means based on our last communication with the LUH team ("For the LUH2 historical data, we use the HYDE "total rice" data and assume that rice is all flooded - either rainfed or irrigated. From that data, and our own estimates of c3ann crops in each grid cell, we compute the fraction of c3ann crops occupied by rice/flooded area"). 

Additionally, I added some details in the creation of the .nc files, which were requested by the LUH team during our interactions for ISIMIP3b, including adding Cropland and Grazing land (pastures+rangelands) in the "States" .nc file and specific namings for datatype,  zname, xname, and yname.

## :wrench: Checklist for PR creator :wrench:

- [x] Label pull request [from the label list](https://github.com/magpiemodel/magpie/labels).
  - **Low risk**: Simple bugfixes (missing files, updated documentation, typos) or changes  in start or output scripts
  - **Medium risk**: Uncritical changes in the model core (e.g. moderate modifications in non-default realizations)
  - **High risk**: Critical changes in model core or default settings (e.g. changing a model default or adjusting a core mechanic in the model)

- [x] Self-review own code
  - No hard coded numbers and cluster/country/region names.
  - The new code doesn't contain declared but unused parameters or variables.
  - [`magpie4`](https://github.com/pik-piam/magpie4) R library has been updated accordingly and backwards compatible where necessary.
  - `scenario_config.csv` has been updated accordingly (important if `default.cfg` has been updated)

- [x] Document changes 
  - Add changes to `CHANGELOG.md`
  - Where relevant, put In-code documentation comments
  - Properly address updates in interfaces in the module documentations
  - run [`goxygen::goxygen()`](https://github.com/pik-piam/goxygen) and verify the modified code is properly documented

- [x] Perform test runs
  - **Low risk**: 
    - Run a compilation check via `Rscript start.R --> "compilation check"`
  - **Medium risk**: 
    - Run test runs via `Rscript start.R --> "test runs"`
    - Check logs for errors/warnings
  - **High risk**:
    - Run test runs via `Rscript start.R --> "test runs"`
    - Check logs for errors/warnings
    - Default run from the PR target branch for comparison
    - Provide relevant comparison plots (land-use, emissions, food prices, land-use intensity,...)

### :chart_with_downwards_trend: Performance changes :chart_with_upwards_trend:
  
NA

## :rotating_light: Checklist for reviewer :rotating_light:

- PR is labeled correctly
- Code changes look reasonable
  - No hard coded numbers and cluster/country/region names.
  - No unnecessary increase in module interfaces
  - model behavior/performance is satisfactory.
- Changes are properly documented
  - `CHANGELOG` is updated correctly
  - Updates in interfaces have been properly addressed in the module documentations
  - In-code documentation looks appropriate
- [x] content review done (at least 1)
- [ ] RSE review done (at least 1)

### CODE REVIEW

**Review by flohump — APPROVED** [review]
The suggested changes make sense to me. Thanks!

File: scripts/output/extra/disaggregation_LUH2.R [hunk 1]
```diff
@@ -6,7 +6,7 @@
 # |  Contact: magpie@pik-potsdam.de
 
 # ------------------------------------------------------------------------------------------------
-# description: Interpolates MAgPIE results to 0.5 degree resolution in LUH2 format for ISIMIP 3b
+# description: Interpolates MAgPIE results to 0.25 degree resolution in LUH2 format for ISIMIP 3b
```
Comment (pfuehrlich-pik) [inline scripts/output/extra/disaggregation_LUH2.R:9]:
```suggestion
# description: Downscale MAgPIE results to 0.25 degree resolution in LUH2 format for ISIMIP 3b
```

Comment (pfuehrlich-pik) [reply scripts/output/extra/disaggregation_LUH2.R:9]: Did other changes in this PR actually change the resolution from 0.5 to 0.25, or was the comment wrong before?

**Review by mishkos — APPROVED** [review]
Looks good. Thx!

File: scripts/output/extra/disaggregation_LUH2.R (hunk 1 above)
Comment (flohump) [reply scripts/output/extra/disaggregation_LUH2.R:9]: the desc was wrong before and is now correct.

**Pull request merged**